.github/workflows/ci.yml

```diff
@@ -0,0 +1,105 @@
+name: CI
+permissions: read-all
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches: [main]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  # Many color libraries just need this to be set to any value, but at least
+  # one distinguishes color depth, where "3" -> "256-bit color".
+  FORCE_COLOR: 3
```

Comment:
we could add `UV_LOCKED: 1` here, so that we don't have to use `--locked` and `--frozen` anymore in the `uv` commands

Reply:
I'm not sure if `--resolution lowest-direct` is the best way to go here actually. Maybe I'm wrong here, but I thought that SPEC 0 applied to the latest patch release of a minor version.